frontend/src/concepts/areas/const.ts

```diff
@@ -0,0 +1,45 @@
+import { StackComponent, SupportedArea, SupportedAreasState } from './types';
+
+export const SupportedAreasStateMap: SupportedAreasState = {
+  [SupportedArea.BYON]: {
+    featureFlags: ['disableBYONImageStream'],
+  },
+  [SupportedArea.CLUSTER_SETTINGS]: {
+    featureFlags: ['disableClusterManager'],
+  },
+  [SupportedArea.CUSTOM_RUNTIMES]: {
+    featureFlags: ['disableModelServing', 'disableCustomServingRuntimes'],
+    reliantAreas: [SupportedArea.MODEL_SERVING],
+  },
+  [SupportedArea.DS_PIPELINES]: {
+    featureFlags: ['disablePipelines'],
+    requiredComponents: [StackComponent.DS_PIPELINES],
+  },
+  [SupportedArea.DS_PROJECTS_VIEW]: {
+    featureFlags: ['disableProjects'],
+  },
+  [SupportedArea.DS_PROJECTS_PERMISSIONS]: {
+    featureFlags: ['disableProjectSharing'],
+    reliantAreas: [SupportedArea.DS_PROJECTS_VIEW],
+  },
+  [SupportedArea.K_SERVE]: {
+    featureFlags: ['disableModelServing'], // TODO: KServe feature flag
```

Comment:
```suggestion
    featureFlags: ['disableKServe'],
```

Reply:
Doesn't exist on main, won't do -- it's a follow up for merging into KServe branch

Reply:
makes sense if it's type safe and doesn't allow it right now